types/persisted-model.d.ts

```diff
@@ -477,6 +478,58 @@ export declare class PersistedModel extends ModelBase {
     callback?: Callback<boolean>,
   ): PromiseOrVoid<boolean>;
 
+  /**
+ * Begin a new transaction.
+ *
+ * A transaction can be committed or rolled back. If timeout happens, the
+ * transaction will be rolled back. Please note a transaction is typically
+ * associated with a pooled connection. Committing or rolling back a transaction
+ * will release the connection back to the pool.
+ *
+ * Once the transaction is committed or rolled back, the connection property
+ * will be set to null to mark the transaction to be inactive. Trying to commit
+ * or rollback an inactive transaction will receive an error from the callback.
+ *
+ * Please also note that the transaction is only honored with the same data
+ * source/connector instance. CRUD methods will not join the current transaction
+ * if its model is not attached the same data source.
+ *
+ * Example:
+ *
+ * To pass the transaction context to one of the CRUD methods, use the `options`
+ * argument with `transaction` property, for example,
+ *
+ * ```js
+ * MyModel.beginTransaction('READ COMMITTED', function(err, tx) {
+  *   MyModel.create({x: 1, y: 'a'}, {transaction: tx}, function(err, inst) {
+  *     MyModel.find({x: 1}, {transaction: tx}, function(err, results) {
+  *       // ...
+  *       tx.commit(function(err) {...});
+  *     });
+  *   });
+  * });
+  * ```
+  *
+  * @param {Object|String} options Options to be passed upon transaction.
+  *
+  * Can be one of the forms:
+  *  - Object: {isolationLevel: '...', timeout: 1000}
+  *  - String: isolationLevel
+  *
+  * Valid values of `isolationLevel` are:
+  *
+  * - Transaction.READ_COMMITTED = 'READ COMMITTED'; // default
+  * - Transaction.READ_UNCOMMITTED = 'READ UNCOMMITTED';
+  * - Transaction.SERIALIZABLE = 'SERIALIZABLE';
+  * - Transaction.REPEATABLE_READ = 'REPEATABLE READ';
+  * @callback {Function} cb Callback function.
+  * @returns {Promise|undefined} Returns a callback promise.
+  */
+  beginTransaction(
+    options?: Options,
+    callback?: Callback,
```

Comment:
Do we know the returned data's type in the callback function? Like https://github.com/strongloop/loopback-datasource-juggler/blob/2b08acf9cb92c65657c93425a1be748775b7ad54/types/persisted-model.d.ts#L42
or it could be anything?

Reply:
Nice catch Janny, I believe it's `Callback<Transaction>`.